editor/src/core/workers/parser-printer/parse-cache-utils.worker.ts

```diff
@@ -33,15 +34,22 @@ function logCacheMessage(parsingCacheOptions: ParseCacheOptions, ...messages: (s
   }
 }
 
+function isArbitraryCodeFile(filename: string): boolean {
+  return filename === ARBITRARY_CODE_FILE_NAME
+}
+
 function stringIdentifiers(filename: string, content: string): (string | object)[] {
-  if (filename === ARBITRARY_CODE_FILE_NAME) {
+  if (isArbitraryCodeFile(filename)) {
     return ['code', [{ content: content }]]
   }
   return [filename]
 }
 
-function shouldSkipCacheForFile(filename: string, parsingCacheOptions: ParseCacheOptions): boolean {
-  return filename === ARBITRARY_CODE_FILE_NAME && !parsingCacheOptions.cacheArbitraryCode
+function shouldUseCacheForFile(filename: string, parsingCacheOptions: ParseCacheOptions): boolean {
```

Comment:
this is the main change - checking also for `isParsableFile` inside `shouldUseCacheForFile`